sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -81,6 +74,43 @@ trait StateStore {
     iterator()
   }
 
+  /**
+   * Return an iterator containing all the key-value pairs in the StateStore. Implementations must
+   * ensure that updates (puts, removes) can be made while iterating over this iterator.
```

Comment:
`Implementations must ensure that updates ...`, but `ReadStateStore` doesn't support puts, removes, right?

Reply:
Yeah right. Let me split `ReadStateStore` and `StateStore`.